src/psfmachine/machine.py

```diff
@@ -502,30 +505,33 @@ def _get_source_mask(
         self._get_uncontaminated_pixel_mask()
 
         # Now we can update the r and phi estimates, allowing for a slight centroid
-        # offset not necessary to take values from Quantity to do .multiply()
-        dx, dy = (
-            self.uncontaminated_source_mask.multiply(self.dra),
-            self.uncontaminated_source_mask.multiply(self.ddec),
-        )
-        dx = dx.data
-        dy = dy.data
-
-        mean_f = np.log10(
-            self.uncontaminated_source_mask.astype(float)
-            .multiply(self.flux[self.time_mask].mean(axis=0))
-            .multiply(1 / self.source_flux_estimates[:, None])
-            .data
-        )
-
-        k = np.isfinite(mean_f)
-        ra_cent = np.average(dx[k], weights=mean_f[k])
-        dec_cent = np.average(dy[k], weights=mean_f[k])
-
-        # re-estimate dra, ddec with centroid shifts, check if sparse case applies.
-        if self.nsources * self.npixels < 1e7:
-            self._create_delta_arrays(centroid_offset=[ra_cent, dec_cent])
-        else:
-            self._create_delta_sparse_arrays(centroid_offset=[ra_cent, dec_cent])
+        # calculate image centroids and correct dra,ddec for offset.
+        if correct_centroid_offset:
+            self._get_centroids()
+            # print(self.ra_centroid_avg.to("arcsec"), self.dec_centroid_avg.to("arcsec"))
+            # re-estimate dra, ddec with centroid shifts, check if sparse case applies.
+            if self.nsources * self.npixels < 1e7:
```

Comment:
Can you add a comment that this is hard coded based on your investigation? Maybe even link to the issue number so we can see the plots you made!